pkg/agent/runner.go

```diff
@@ -1660,7 +1665,12 @@ func (r *Runner) doInformantDownscale(ctx context.Context, logger *zap.Logger, t
 		return nil, fmt.Errorf("%s: InformantServer is not set (this should not occur after startup)", msg)
 	}
 
-	resp, err := server.Downscale(ctx, logger, to)
+	downscale := server.Downscale
+	if server.informantIsMonitor {
```

Comment:
Same question here about "can this change because we're not holding the lock?" (same for upscale as well)